### PR TITLE
completed

### PR DESCRIPTION
**Pull request by xialiu1988**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: Data-Structures/src/main/java/Hashtable/Hashtable.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    public Hashtable(int size){
+        this.size=size;
+        arr=new Llist[size];
+    }
+
+    //add value to hashtable
+    public void add(String key,String value){
+   int index=hash(key);
+        if (arr[index] == null)
```
Comment (evanslaton) [inline Data-Structures/src/main/java/Hashtable/Hashtable.java]: Remember that in Java you should but opening `{` on the same line as the thing it's opening!

File: Data-Structures/src/main/java/Hashtable/Hashtable.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        int index = hash(key);
+        if (arr[index] != null)
+        {
+            return arr[index];
+        }
+
+        return null;
+    }
+
+    public boolean contains(String key)
```
Comment (evanslaton) [inline Data-Structures/src/main/java/Hashtable/Hashtable.java]: This method won't work properly if there was a collision. If arr[index] has a LinkedList then this method will return true even if the actual key that's passed in isn't there.
